spatial-partition/src/test/java/com/iluwatar/spatial/partition/SpatialPartitionTest.java

```diff
@@ -0,0 +1,54 @@
+/**
+ * The MIT License
+ * Copyright (c) 2014 Ilkka Seppälä
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package com.iluwatar.spatial.partition;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Created by kristof.szoke on 2017-05-08.
+ */
+public class SpatialPartitionTest {
+
+  private Grid grid;
+
+  private Unit unit;
+
+  private int x;
+
+  private int y;
+
+  @Before
+  public void setUp() {
+    grid = new Grid();
+    unit = new Unit(grid, x, y);
+  }
+
+  @Test
+  public void initialiseEmpty() {
+    grid.clearGrid();
+    assertEquals(grid.getCells().length, 10);
+  }
+}
```

Comment:
The unit tests are insufficient. Add test for running the application. Add tests for `Grid` and `Unit` essential methods.